### PR TITLE
drivers/a*: model modules in Kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
Continuing with the module dependency modelling in Kconfig, this is the first of a series of PRs that will start modelling the high-level drivers.  Given the amount of drivers I will be splitting this into multiple PRs. I've started in alphabetical order, this contains most of the one starting with `a`. Some network devices are not in this PR given that many test applications bring the GNRC stack, and that's not yet modelled.

### Testing procedure
- Check that the dependencies are correctly modelled, comparing to the current Makefiles.
- Check that the test applications of the correspondent drivers are added to the list of the CI
- Green CI

### Issues/PRs references
None

### CODE REVIEW

File: drivers/at24cxxx/Kconfig [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    bool "AT24C128"
+    select MODULE_AT24CXXX
+
+config MODULE_AT24C256
+    bool "AT24C256"
+    select MODULE_AT24CXXX
+
+config MODULE_AT24C512
+    bool "AT24C512"
+    select MODULE_AT24CXXX
```
Comment (benpicco) [inline drivers/at24cxxx/Kconfig:38]: It actually supports [more than that](https://github.com/RIOT-OS/RIOT/blob/master/drivers/at24cxxx/include/at24cxxx_defines.h#L328) - and even more if those parameters are supplied by the user.

Comment (leandrolanzieri) [reply drivers/at24cxxx/Kconfig:38]: Ah thanks, I had a hard time trying to figure which models were supported, will add them!

File: tests/driver_at24mac/app.config.test [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# this file enables modules defined in Kconfig. Do not use this file for
+# application configuration. This is only needed during migration.
+CONFIG_MODULE_AT24C256=y
```
Comment (benpicco) [inline tests/driver_at24mac/app.config.test:3]: seems unrelated

Comment (leandrolanzieri) [reply tests/driver_at24mac/app.config.test:3]: Well, the `at24mac` needs some implementation of `at24cxxx`

Comment (benpicco) [reply tests/driver_at24mac/app.config.test:3]: No it doesn't, only if you want to use the EEPROM functionality.
The ID functionality is separate.

Comment (leandrolanzieri) [reply tests/driver_at24mac/app.config.test:3]: I don't understand, here the module adds the dependency:
https://github.com/RIOT-OS/RIOT/blob/a3e4c1eca5bf40215174dcc4e97795d5487fbcbc/drivers/at24mac/Makefile.dep#L1

Also, appears in the pulled modules of the test application:
```
at24cxxx
at24mac
auto_init
auto_init_xtimer
board
core
core_init
core_msg
core_panic
cortexm_common
cortexm_common_periph
cpu
div
newlib
newlib_nano
newlib_syscalls_default
periph
periph_common
periph_gpio
periph_i2c
periph_init
periph_init_gpio
periph_init_i2c
periph_init_pm
periph_init_timer
periph_init_uart
periph_pm
periph_timer
periph_uart
pm_layered
sam0_common_periph
stdio_uart
sys
xtimer
```

Comment (benpicco) [reply tests/driver_at24mac/app.config.test:3]: `at24mac` is *also* an implementation of `at24cxxx`

Comment (leandrolanzieri) [reply tests/driver_at24mac/app.config.test:3]: Hmm, ok, will try to model that

File: drivers/at24cxxx/Kconfig (hunk 1 above)
Comment (leandrolanzieri) [reply drivers/at24cxxx/Kconfig]: I've added the modules in the list. Later we can improve this by adding one more option where the user can manually enter the parameters, but for that we need to make them configurable first.

File: tests/driver_at24mac/app.config.test (hunk 2 above)
Comment (leandrolanzieri) [reply tests/driver_at24mac/app.config.test]: Done, now this module selects `at24cxxx` as well

File: tests/driver_at/app.config.test [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# this file enables modules defined in Kconfig. Do not use this file for
+# application configuration. This is only needed during migration.
+# dependencies of AT driver
+CONFIG_MODULE_ISRPIPE=y
+CONFIG_MODULE_ISRPIPE_READ_TIMEOUT=y
```
Comment (fjmolinas) [inline tests/driver_at/app.config.test:5]: Why are these specified here? Arent they part of the driver dependencies?

Comment (leandrolanzieri) [reply tests/driver_at/app.config.test:5]: Yes, that's why they have to be enabled to be able to use the driver

File: sys/event/Kconfig [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+if MODULE_EVENT
+
+config MODULE_EVENT_CALLBACK
+    bool "Support for callback-with-argument event type"
+
+menuconfig MODULE_EVENT_THREAD
+    bool "Support for event handler threads"
+    help
+        There are three threads of different priorities that can be enabled.
```
Comment (fjmolinas) [inline sys/event/Kconfig:27]: Shouldn't this depend on the `thread` module as well.

File: .murdock [hunk 5]
```diff
@@ -3,7 +3,11 @@
 : ${TEST_BOARDS_AVAILABLE:="esp32-wroom-32 samr21-xpro"}
 : ${TEST_BOARDS_LLVM_COMPILE:="iotlab-m3 native nrf52dk mulle nucleo-f401re samr21-xpro slstk3402a"}
 # we can't use '-' in the variable names, convert them to '_' to add new boards
-: ${TEST_KCONFIG_samr21_xpro:="examples/hello-world tests/periph_*"}
+: ${TEST_KCONFIG_samr21_xpro:="examples/hello-world tests/periph_*
+tests/driver_ad7746 tests/driver_adcxx1c tests/driver_ads101x tests/driver_adt101x
+tests/driver_adt7310 tests/driver_adxl345 tests/driver_aip31068 tests/driver_apa102
+tests/driver_apds99xx tests/driver_apds99xx_full tests/driver_at tests/driver_at24cxxx
+tests/driver_at24mac tests/driver_at25xxx tests/driver_at30tse75x tests/driver_ata8520e"}
```
Comment (fjmolinas) [inline .murdock:10]: Why not `tests/driver_a*`?

File: sys/event/Kconfig (hunk 4 above)
Comment (leandrolanzieri) [reply sys/event/Kconfig:27]: Is that a module?

**Review by fjmolinas — COMMENTED** [review]
Looks good, I added a couple of comments. If possible can you when squashing re-order commits so all commits adding `sys/*/Kconfig` are at the beginning?

File: .murdock (hunk 5 above)
Comment (leandrolanzieri) [reply .murdock:10]: That was the intention at first, but could not get around all the needed things that at86rf215 needs in the test application without bloating this. 

> Some network devices are not in this PR given that many test applications bring the GNRC stack, and that's not yet modelled.

File: sys/event/Kconfig (hunk 4 above)
Comment (fjmolinas) [reply sys/event/Kconfig:27]: Ah no it isn't sorry

File: tests/driver_at/app.config.test (hunk 3 above)
Comment (fjmolinas) [reply tests/driver_at/app.config.test:5]: Sorry dumb question, a bit too used to make.

File: .murdock (hunk 5 above)
Comment (fjmolinas) [reply .murdock:10]: Ahh Ok.

**Comment by fjmolinas** [comment]
> That was the intention at first, but could not get around all the needed things for at86rf215 in the test application without bloating this.

Then please squash, and i you agree re-order the `sys/*/Kconfig` commits :)

**Comment by leandrolanzieri** [comment]
> > That was the intention at first, but could not get around all the needed things for at86rf215 in the test application without bloating this.
> 
> Then please squash, and i you agree re-order the `sys/*/Kconfig` commits :)

Squashed and reordered !

**Review by fjmolinas — APPROVED** [review]
ACK.

**Comment by leandrolanzieri** [comment]
Rebased to resolve conflicts with `.murdock`

**Comment by leandrolanzieri** [comment]
And again

File: sys/event/Kconfig [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+if MODULE_EVENT_THREAD
+
+config MODULE_EVENT_THREAD_LOW
```
Comment (leandrolanzieri) [inline sys/event/Kconfig:31]: I just noticed that the actual name of this symbol should be `MODULE_EVENT_THREAD_LOWEST`, will push a fix

Comment (leandrolanzieri) [reply sys/event/Kconfig]: Done

**Comment by leandrolanzieri** [comment]
Everything looks good
